pandas/tests/frame/methods/test_combine_first.py

```diff
@@ -398,6 +398,21 @@ def test_combine_first_string_dtype_only_na(self, nullable_string_dtype):
         ).set_index(["a", "b"])
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "wide_val, dtype",
+        (
+            (1666880195890293744, "uint64"),
+            (-1666880195890293744, "int64"),
+        ),
+    )
+    def test_combine_first_preserve_precision(self, wide_val, dtype):
```

Comment:
looks like you made changes to both combine and combine_first, but only a test for combine_first.  can you test the fixed bug in combine?